### PR TITLE
ML Translation

### PR DESCRIPTION
**Pull request by elaineliucs**
Utilized the Cloud Translation API to allow users to translate text from one language to another. 

1) Created a helper function in MessageServlet.java called translateMessages that takes a List of Message instances, iterates through the list, and translates the text to the specified language. 
2) Modified the doGet() function, which gets the target language from the query string parameter and calls the helper function translateMessages.
3) Modified the fetchMessages() function in user-page-loader.js to get the parameter language from the URL and pass it into the URL used to fetch messages. 
4) Added a list element to the page by modifying the user-page.html file. 
5) Created a function buildLanguageLinks() in user-page-loader.js to add language links so that the user doesn't have to type in the language parameter manually. 

ML Translation

### CODE REVIEW

**Review by sundyx — APPROVED** [review]
Everything looks correct to me!

File: src/main/webapp/user-page.html [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <link rel="stylesheet" href="/css/main.css">
     <link rel="stylesheet" href="/css/user-page.css">
     <script src="/js/user-page-loader.js"></script>
+    <script src="/js/navigation-loader.js"></script>
```
Comment (jonathantanmy) [inline src/main/webapp/user-page.html:26]: After some investigation, I think that Elaine added this line because the user page was missing navigation links and not really because the ML Translation feature needed it. Personally, I include such cleanups only if I was already editing that line (if not, I would put it in a separate PR). No change needed in this PR, but something else to think about.

**Review by jonathantanmy — APPROVED** [review]
